### PR TITLE
More test and optimization

### PR DESCRIPTION
**Pull request by miriamhaenle**
Hello!
My PR request mainly consists of improved tests and implementing feedback from the code review :) 
Thank you for the review!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -119,11 +116,14 @@ export default function App() {
   function updateCarbonFootprint(value) {
     setCarbonFootprint([...carbonFootprint, value])
   }
-  function updateFootprintPerTransportationType(type, sum) {
+  function updateFootprintPerTransportationType(
+    transportationTypeToUpdate,
```
Comment (sv-widegreen) [inline src/App.js:120]: 🥇

File: src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  })
+  it('should map empty strings if transportation sum is empty', () => {
+    expect(
+      mapFootprintPerTransportTypeToDonutChartData({
+        transportationType: 'car',
+        footprintSum: '',
+      })
+    ).toEqual({ label: '', y: '' })
+  })
+  it('should map empty stringsp if no sum is send', () => {
```
Comment (sv-widegreen) [inline src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js:28]: ein kleiner Rechtschreibfehler in dieser Zeile und eine Frage - hier könnten doch zwei Gründe das expect bestätigen, einmal wonach du testest (keine sum wird gesendet), aber auch da der transportationType auch ein empty string ist (also wie in Test Zeile 12) müsste ja zu emtpy strings führen? Vielleicht machst du hier keinen empty string in Zeile 31 sondern Daten die eigentlich durchgehen müssten, um wirklich sicher zu sein.

File: src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    expect(
+      mapFootprintPerTransportTypeToDonutChartData({
+        transportationType: '',
+      })
+    ).toEqual({ label: '', y: '' })
+  })
+  it('should map empty strings if no transportation Type is send', () => {
+    expect(
+      mapFootprintPerTransportTypeToDonutChartData({
+        footprintSum: '',
```
Comment (sv-widegreen) [inline src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js:38]: Hier das gleiche wie oben?

File: src/services/calculateCarbonEmission.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        carbonFootprint: '',
+      },
+    })
+    const carbonFootprint = await calculateCarbonEmission({
+      distance: '-3',
+      transportationType: 'anyCar',
+    })
+    expect(carbonFootprint).toEqual(0)
+  })
+})
```
Comment (sv-widegreen) [inline src/services/calculateCarbonEmission.test.js:136]: kann man auch was testen falls der transportation type ein anderer ist der nicht erwartet werden würde oder ist nur die distance die wichtige Größe die übermittelt werden muss?

File: src/services/calculateFootprintPerTransportationType.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  { transportationTypeToUpdate, footprintSum }
+) {
+  if (!transportationTypeToUpdate) {
+    return currentState
+  }
+  if (!footprintSum) {
+    return currentState
+  }
+  if (transportationTypeExists(currentState, transportationTypeToUpdate)) {
+    return stateWithUpdatForExistingTransportationType(currentState, {
```
Comment (sv-widegreen) [inline src/services/calculateFootprintPerTransportationType.js:12]: Update mit 'e'

File: src/services/calculateFootprintPerTransportationType.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+function transportationTypeExists(currentState, transportationTypeToUpdate) {
+  return currentState.find(
+    (transportationType) =>
+      transportationType.transportationType === transportationTypeToUpdate
+  )
+}
+
+function stateWithUpdatForExistingTransportationType(
```
Comment (sv-widegreen) [inline src/services/calculateFootprintPerTransportationType.js:31]: ...dann hier auch mit e 🤓

File: src/services/calculateFootprintPerTransportationType.test.js [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    expect(
+      calculateFootprintPerTransportionType(
+        [
+          {
+            transportationType: 'car',
+            footprintSum: 200,
+          },
+          { transportationType: 'bus', footprintSum: 400 },
+        ],
+        { transportationTypeToUpdate: '' }
```
Comment (sv-widegreen) [inline src/services/calculateFootprintPerTransportationType.test.js:114]: Hier vielleicht auch lieber was mitschicken beim type um zu checken dass es auch wirklich an der missing data liegt

File: src/services/handleStorage.js [hunk 8]
```diff
@@ -1,19 +1,23 @@
 import firebase from 'firebase'
-import { db } from '../../firebase/index'
+import { db } from '../firebase/index'
 
 export async function saveToStorage(userId, key, dataSet) {
   const userDoc = db.collection(key).doc(userId)
```
Comment (sv-widegreen) [inline src/services/handleStorage.js:5]: Und ich dachte gerade, was macht denn der Arzt hier? 😂  👨‍⚕️  Ist das häufig so benannt?

File: src/services/handleStorage.test.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import { getFromStorage, saveToStorage } from './handleStorage'
+
+describe('save to storage', () => {
+  it('should save dataset to storage', () => {
+    const mockSaveToStorage = jest.fn(saveToStorage)
```
Comment (sv-widegreen) [inline src/services/handleStorage.test.js:5]: Kann man hier vielleicht Zeilen 5 bis 11 in ein before Each unters describe packen?

File: src/services/handleStorage.test.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    const DATASET = { foo: 'bar' }
+
+    mockSaveToStorage(USERID, KEY, DATASET)
+    expect(mockSaveToStorage).toHaveBeenCalledWith(USERID, KEY, DATASET)
+  })
+})
+
+describe('get from storage', () => {
+  it('should get dataset from storage', () => {
+    const mockGetFromStorage = jest.fn(getFromStorage)
```
Comment (sv-widegreen) [inline src/services/handleStorage.test.js:29]: Hier auch, Zeilen 29 bis 33 in ein beforeEach?

File: src/services/handleStorage.js (hunk 8 above)
Comment (miriamhaenle) [reply src/services/handleStorage.js:5]: haha, ja so stand es in den Dokus und bei Joe 😁

File: src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js (hunk 2 above)
Comment (miriamhaenle) [reply src/components/DonutChart/mapFootprintPerTransportTypeToDonutChartData.test.js:28]: Ah du hast Recht! Danke :)

File: src/services/calculateFootprintPerTransportationType.js (hunk 6 above)
Comment (miriamhaenle) [reply src/services/calculateFootprintPerTransportationType.js:31]: Gutes Auge 🤓
